client/src/features/post/postComponenet.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { FunctionComponent } from "react";
+import Description from "../../pages/create-post/description";
+import ChooseSong from "../../pages/create-post/choose-song";
+import { FaEdit, FaTrashAlt, FaPlay, FaRegHeart, FaComment, FaReply} from "react-icons/fa";
+import { RiStackFill, RiPlayList2Fill } from "react-icons/ri";
+
+
+export interface SongInfo {
+    name : string
+    albumCoverPath: string
+    artistName: string
+
+}
+
+interface PostComponentProps {
+    username: string
+    profilePath: string
+    authorCaption: string
+    song: SongInfo
+
+
+
+}
+
+/*
+
+    Example PostComponent:
+
+     <PostComponent username="McLovin" profilePath="/2357380.jpg" authorCaption="I never lie, including about my age" song={{
+        albumCoverPath: "/final1.png",
+        name: "The Man Who Never Lied",
+        artistName: "Maroon 5"
+      }}/>
+
+
+
+
+
+*/
+
+
+
+
+
+
+
+//PostComponent Feature
+//TODO: 
+// Figure out how to contrain comment data structure in the props of this component
+// Figure out how to expand PostComponent to show comments
+// Button Handling for all buttons
+
+function PostComponent(prop: PostComponentProps) {
+
+    return (
+        <div className= "flex flex-col h-[346px] w-[600px] bg-[#334155] rounded-lg">
+
+            <div className="flex justify-between items-center h-[75px]" >
+
+                <div className="flex ">
+                    <img src={prop.profilePath} className= "rounded-full w-[45px] h-[45px] ml-5"/>
+                    <h3 className = " flex text-[15px] text-[#FEFCE8] ml-3 items-center font-bold">{prop.username}</h3>
+
+                </div>
+
+                <div>
+                    <button> <FaEdit className="text-green-500 mr-5 h-[20px] w-[20px]"></FaEdit></button>
+                    <button><FaTrashAlt className="text-green-500 mr-5 h-[20px] w-[20px]"></FaTrashAlt></button>
+
+                </div>
+            </div>
+
+            <div className="bg-[#475569] h-[3px]"></div>
+
+            <div className="flex bg-[#1E293B] h-[125px] items-center justify-between">
+
+                <div className="flex items-center">
+                    <img src={prop.song.albumCoverPath} className='h-[75px] w-[75px] rounded-lg ml-8'></img>
```

Comment:
Use the existing MusicItem component, and these icon buttons to create a resusable PlayableMusicItem component. Anything that will be used or has a lot of extra logic should be a separate component